src/JiraSecurityIssue.php

```diff
@@ -277,7 +277,7 @@ public function exists(): ?string
         return null;
     }
 
-    public function userNameByEmail(string $email): ?string
+    public function userNameByEmail(string $email): ?User
```

Comment:
Should be renamed to `userByEmail` (some would argue for `getUserByEmail()`, but I agree with Fowler that `user()`/`setUser()` is nicer than `getUser()`/`setUser()`).

Reply:
And actually it is not even necessarily an email we lookup. It's just a query string...

Reply:
`userDataByEmail` -> `findUsers`

`userNameByEmail` -> `findFirstUser` ?

Or refactor everything to use `findUsers`, the second function really doesn't do much anymore.

Reply:
We'll go with `findUser(string $query) ` and both functions merged.